js/embed_media_back_button.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file
+ * Adds the back button for embeded media select modal in CKEditor.
+ * Everytime the user goes to different page within the media embed select modal
+ * it will save the filter values to retrive the search result when user
+ * comes back to the first page.
+ */
+
+ (function ($, Drupal) {
+  'use strict';
+
+  Drupal.behaviors.tide_media_browser_iframe = {
+    attach: function (context, settings) {
+      if (!window.CKEDITOR) {
+        return;
+      }
+      let iframePath, itemName, licenseType, mediaType, published, site
+      const iframeName = 'entity_browser_iframe_tide_media_browser_iframe'
+      const homePagePath = '/entity-browser/iframe/tide_media_browser_iframe'
+      const backButton = $('#entity-embed-dialog-tide-media-browser-home')
+      $(document).ready(function () {
+        // Remove the name field value on modal first load if any.
+        localStorage.removeItem("tideMediaBrowsernameFilterVal");
+        $("iframe").on("load", function() {
+          if ($("iframe") && $("iframe").length > 0) {
```

Comment:
there can be multiple iframe loaded in a single page, so this will make sure if there are multiple iframe it will check it is targeting the correct one